src/prepared_statement.rs

```diff
@@ -0,0 +1,174 @@
+
+use frame::{Frame, Flag};
+use IntoBytes;
+use frame::parser::parse_frame;
+use authenticators::Authenticator;
+use error;
+use transport::CDRSTransport;
+use consistency::Consistency;
+use client::Session;
+use query::QueryParamsBuilder;
+use types::value::Value;
+use frame::frame_response::ResponseBody;
+use std::collections::BTreeMap;
+use std::convert::Into;
+use byteorder::{BigEndian, WriteBytesExt};
+
+#[derive(Debug)]
+pub struct PreparedStatement {
+    tracing: Option<bool>,
+    warnings: Option<bool>,
+    consistency: Consistency,
+    with_names: bool,
+    page_size: Option<i32>,
+    paging_state: Option<i32>,
+    serial_consistency: Option<Consistency>,
+    timestamp: Option<i64>,
+    frame: Option<Frame>,
+    query_markers: BTreeMap<u32, Value>,
+}
+
+impl Default for PreparedStatement {
+    fn default() -> PreparedStatement {
+        let query_markers = BTreeMap::new();
+
+        PreparedStatement {
+            tracing: None,
+            warnings: None,
+            consistency: Consistency::One,
+            with_names: false,
+            page_size: None,
+            paging_state: None,
+            serial_consistency: None,
+            timestamp: None,
+            frame: None,
+            query_markers: query_markers,
+        }
+    }
+}
+
+impl PreparedStatement {
+    /// `index` represent the position of  marker `?` in the query
+    pub fn set_string(&mut self, index: u32, val: &'static str) -> error::Result<()> {
+        self.query_markers.insert(index, val.to_string().into());
+        Ok(())
+    }
+
+    // /// `index` represent the position of  marker `?` in the query
+    // /// date should be in the format epochtime
+    // not yet working
+    // pub fn set_date(&mut self, index: u32, val: i64) -> error::Result<()> {
+    //     let mut wtr = vec![];
+    //     match wtr.write_i64::<BigEndian>(val) {
+    //         Ok(_) => {
+    //             self.query_markers.insert(index, Value::new_normal(wtr.clone()));
+    //             Ok(())
+    //         }
+    //         Err(err) => return Err(error::Error::from(err)),
+    //     }
+    // }
+
+    /// `index` represent the position of marker `?` in the query
+    pub fn set_int(&mut self, index: u32, val: i32) -> error::Result<()> {
```

Comment:
I haven't completed it yet !!! Would need methods for setdate set list set map settuple , this is the first cut we need to add the rest and plus validate method as well so that all the indexes are indeed provided and each value set is same as it in backend ; like Client should nt set a string for varint db type